cleanup-anaconda: Fix and enhance the whole logic

ajelinski opened the pull request:
Fixes and adds a few new things to the cleanup script.

A clean summary is printed to the log file which at the end is printed to the stdout.
I've added it as the output is messy because of `set -x` (but it's best to keep it for extra details).
I'll also upload it as an artifact in the `master-package` job so we can easily track these removals if there are any doubts.

mithro commented:
Generally LGTM. If @PiotrZierhoffer is happy merge away.

PiotrZierhoffer commented on cleanup-anaconda.sh:
```diff
@@ -3,35 +3,90 @@
+    cat $log_file
+    end_section
+}
+trap print_log EXIT
```
nice touch

PiotrZierhoffer commented on cleanup-anaconda.sh:
```diff
@@ -3,35 +3,90 @@
+#packages will be removed if their label is older than 'ago'
 ago="7 days ago"
+#packages won't be removed if they were downloaded more than 'download_threshold' times
+download_threshold=30
```
I think we should start with 0, actually.

ajelinski replied:
This might cause some packages to never be cleaned.
For example, `isl` for Linux seems to have 24 downloads right after the CI is finished: https://anaconda.org/LiteX-Hub/isl/files

This is an extreme example but there are at least a few packages with 3+ downloads so I'd go for at least 5-10.
What do you think?

PiotrZierhoffer commented on cleanup-anaconda.sh:
```diff
@@ -3,35 +3,90 @@
+        ndownloads=$(echo "$metadata" | grep -oP '(?<=ndownloads : )[0-9]+')
+        if [[ $ndownloads -gt $download_threshold ]]
+        then
+            echo "In use:  $package" | tee -a $log_file
```
can we log `ndownloads` as well?

PiotrZierhoffer commented:
The conclusion is to:

- keep 30 as a limit
- increase the timeout to 30 days

ajelinski commented:
Changed the timeout and added `ndownloads` as well as labels to be printed in a summary for all packages.

Now at the end such a summary will be printed (the actual removing was disabled; the download threshold was 0):
```
RESULT   LABELS                 NDOWNLOADS  PACKAGE
In use   ['ci-HEAD-480007789']  1           linux-64/capnproto-0.8.0-20210112_105211.tar.bz2
In use   ['ci-HEAD-480342687']  2           linux-64/icestorm-v0.0_719_g792cef0-20210112_134626.tar.bz2
In use   ['ci-HEAD-480342687']  4           linux-64/iverilog-v0_8_5850_g540555fc-20210112_134626.tar.bz2
In use   ['ci-HEAD-480342687']  1           linux-64/prjtrellis-v1.0_176_gf93243b-20210112_134626.tar.bz2
Removed  ['ci-HEAD-480342687']  0           linux-64/quicklogic-vtr-v8.0.0_3270_gaec608ff6-20210112_134626.tar.bz2
Removed  ['ci-HEAD-480342687']  0           linux-64/quicklogic-yosys-plugins-v1.2.0_11_g21045a9-20210112_134626.tar.bz2
Removed  ['ci-HEAD-480342687']  0           linux-64/quicklogic-yosys-v0.8.0_59_gadff579e-20210112_134626.tar.bz2
Removed  ['ci-HEAD-480342687']  0           linux-64/slang-v0.5_197_g24743e22-20210112_134626.tar.bz2
```

PiotrZierhoffer commented on cleanup-anaconda.sh:
```diff
@@ -3,35 +3,99 @@
+            print_result "Removed"
+        else
+            # This might be caused by a simultaneously running cleanup script
+            print_result "Error"
```
`Error removing`, otherwise good to merge and bump

ajelinski replied:
Used `Rm error` to not widen the result column too much.
